db/cache_test.go

```diff
@@ -0,0 +1,173 @@
+package db
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/pokt-network/poktroll/pkg/polylog/polyzero"
+	"github.com/stretchr/testify/require"
+	gomock "go.uber.org/mock/gomock"
+
+	"github.com/buildwithgrove/path/user"
+)
+
+func Test_GetGatewayEndpoint(t *testing.T) {
+	tests := []struct {
+		name       string
+		endpointID user.EndpointID
+		mockReturn map[user.EndpointID]user.GatewayEndpoint
+		expected   user.GatewayEndpoint
+		found      bool
+	}{
+		{
+			name:       "should return gateway endpoint when found",
+			endpointID: "endpoint_1",
+			mockReturn: getTestGatewayEndpoints(),
+			expected:   getTestGatewayEndpoints()["endpoint_1"],
+			found:      true,
+		},
+		{
+			name:       "should return different gateway endpoint when found",
+			endpointID: "endpoint_2",
+			mockReturn: getTestGatewayEndpoints(),
+			expected:   getTestGatewayEndpoints()["endpoint_2"],
+			found:      true,
+		},
+		{
+			name:       "should return false when gateway endpoint not found",
+			endpointID: "endpoint_3",
+			mockReturn: getTestGatewayEndpoints(),
+			expected:   user.GatewayEndpoint{},
+			found:      false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			c := require.New(t)
+			ctrl := gomock.NewController(t)
+
+			mockDB := NewMockDBDriver(ctrl)
+			mockDB.EXPECT().GetGatewayEndpoints(gomock.Any()).Return(test.mockReturn, nil)
+
+			cache, err := NewUserDataCache(mockDB, time.Minute, polyzero.NewLogger())
+			c.NoError(err)
+
+			gatewayEndpoint, found := cache.GetGatewayEndpoint(test.endpointID)
+			c.Equal(test.found, found)
+			c.Equal(test.expected, gatewayEndpoint)
+		})
+	}
+}
+
+func Test_cacheRefreshHandler(t *testing.T) {
+	tests := []struct {
+		name       string
+		mockReturn map[user.EndpointID]user.GatewayEndpoint
+		expected   map[user.EndpointID]user.GatewayEndpoint
```

Comment:
Ditto. In general, try to separate inputs and expected outputs (with better naming) so it's easier for other developers to understand.

Please update the tests too. A bit of a NIT but ty